ALLOW_WARNINGS=1 for Raspbian build instructions

danielmewes opened the pull request:
As suggested by @mogren https://github.com/rethinkdb/rethinkdb/issues/4541#issuecomment-138243806

dalanmiller commented:
@danielmewes Are you sure we wanted to merge this? I thought we were still uncertain of `ALLOW_WARNINGS=1` on the pre-RBPi2 models?

danielmewes commented:
@dalanmiller I think as long as it's required on at least some ARM platforms, this is better than not adding it. If we introduced some new ARM-specific problem that was revealed in a compiler warning, this increases the chance that we're going to miss it because users probably won't report it. However that's better than having a compilation guide up that doesn't work for a lot of users I think.

Once we've addressed all current issues that cause warnings on some ARM platforms, we can remove the `ALLOW_WARNINGS=1` again.
